python/lsst/daf/butler/dimensions/_records.py

```diff
@@ -490,6 +538,86 @@ def toDict(self, splitTimespan: bool = False) -> dict[str, Any]:
                 results["datetime_end"] = timespan.end
         return results
 
+    def serialize_key_value(self) -> SerializedKeyValueDimensionRecord:
+        """Serialize this record to a `list` that can be sliced into a key
+        (data ID values) / value (everything else) pair.
```

Comment:
If anybody has thoughts on better nomenclature for this new `DimensionRecord` serialization mode (it needs to be distinguished from the old one somehow), I'd love to hear it, because I'm not thrilled with what I've come up with so far.

Reply:
It's still a single DimensionRecord? not effectively `list[DimensionRecord]`?

Reply:
Yes, here.  Serializing a collection of dimension records is over on `DimensionRecordSet.serialize_records` and it delegates to this.